src/coprocessor/circom.rs

```diff
@@ -0,0 +1,159 @@
+//! # Usage of circom coprocessors.
+//!
+//! See `examples/circom.rs` for a quick example of how to declare a circom coprocessor.
+
+#[cfg(not(target_arch = "wasm32"))]
+pub mod non_wasm {
+    use core::fmt::Debug;
+    use std::{fs::read_dir, path::PathBuf};
+
+    use anyhow::{bail, Result};
+    use bellperson::{ConstraintSystem, SynthesisError};
+    use circom_scotia::r1cs::CircomConfig;
+
+    use crate::{
+        circuit::gadgets::{
+            circom::CircomGadget,
+            data::GlobalAllocations,
+            pointer::{AllocatedContPtr, AllocatedPtr},
+        },
+        coprocessor::{CoCircuit, Coprocessor},
+        field::LurkField,
+        ptr::Ptr,
+        store::Store,
+    };
+
+    /// TODO: fix this duplication with the one in `cli::paths`
+    fn circom_dir() -> PathBuf {
+        home::home_dir()
+            .expect("no home directory")
+            .join(".lurk/circom")
+    }
+
+    fn print_error(name: &str, avaliable: Vec<String>) -> Result<()> {
+        let avaliable = avaliable.join("\n    ");
+        bail!(
+            "
+error: no circom gadget named `{name}`.
+Available circom gadgets:{avaliable}
+
+If you want to setup a new circom gadget `{name}`, run
+    `lurk coprocessor --name {name} <{}_FOLDER>`",
```

Comment:
what should I expect to find in the NAME_FOLDER for this to be valid?

Reply:
Expanded